### PR TITLE
v1.18.0 Updates

### PR DESCRIPTION
**Pull request by artificial-aidan**
- Add support for the newly added `HTTPProxy.Spec.IngressClassName`
- Update permissions documentation.

### CODE REVIEW

File: docs/usage.md [hunk 1]
```diff
@@ -31,8 +31,8 @@ To disable CRD creation, specify `crds` command-line flag or `CP_CRDS` environme
 In a normal setup, Contour has a `type=LoadBalancer` Service to expose its Envoy pods to Internet.
 By specifying `service-name`, contour-plus can identify the global IP address for FQDNs in HTTPProxy.
 
-If `ingress-class-name` is specified, contour-plus watches only HTTPProxy annotated by `kubernetes.io/ingress.class=<ingress-class-name>` or `projectcontour.io/ingress.class=<ingress-class-name>`.  
-**If both `kubernetes.io/ingress.class=<ingress-class-name>` and `projectcontour.io/ingress.class=<ingress-class-name>` are specified and those values are different, then contour-plus doesn't watch the resource.**
+If `ingress-class-name` is specified, contour-plus watches only HTTPProxy annotated by `kubernetes.io/ingress.class=<ingress-class-name>`, `projectcontour.io/ingress.class=<ingress-class-name>` or with the `HTTPProxy.Spec.IngressClassName` field that matches the given `ingress-class-name`.
+**If both `kubernetes.io/ingress.class=<ingress-class-name>` and `projectcontour.io/ingress.class=<ingress-class-name>` and `HTTPProxy.Spec.IngressClassName` are specified and those values are different, then contour-plus doesn't watch the resource.**
```
Comment (ysksuzuki) [inline docs/usage.md:35]:
```suggestion
**If `kubernetes.io/ingress.class=<ingress-class-name>` , `projectcontour.io/ingress.class=<ingress-class-name>` and `HTTPProxy.Spec.IngressClassName` are all specified and those values are different from the given `ingress-class-name`, then contour-plus doesn't watch the resource.**
```

**Review by ysksuzuki — CHANGES_REQUESTED** [review]

**Review by ysksuzuki — APPROVED** [review]
Thank you for your contribution!

**Comment by artificial-aidan** [comment]
> Thank you for your contribution!

What's the process for a new release? Is it automatic?

**Comment by ysksuzuki** [comment]
> > Thank you for your contribution!
> 
> What's the process for a new release? Is it automatic?

We update the changelog and push a tag as described here.
https://github.com/cybozu-go/contour-plus/blob/main/RELEASE.md

Do you want the new release right now? If so, I will release v0.6.4.

**Comment by artificial-aidan** [comment]
> > > Thank you for your contribution!
> > 
> > What's the process for a new release? Is it automatic?
> 
> We update the changelog and push a tag as described here.
> https://github.com/cybozu-go/contour-plus/blob/main/RELEASE.md
> 
> Do you want the new release right now? If so, I will release v0.6.4.

That would be great!

**Comment by ysksuzuki** [comment]
Released v0.6.4
https://github.com/cybozu-go/contour-plus/releases/tag/v0.6.4

**Pull request merged**